src/rust/engine/src/externs/interface.rs

```diff
@@ -943,106 +941,116 @@ async fn workunit_to_py_value(
   core: &Arc<Core>,
   session: &Session,
 ) -> CPyResult<Value> {
-  use std::time::UNIX_EPOCH;
+  let mut dict_entries = {
```

Comment:
The diff is bigger than it looks. The only change is putting all the setup of `dict_entries` so that the GIL is dropped at the end of the block (before an async call), along with passing `py` to everything.